File: 백준/Silver/2164. 카드2/카드2.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        int N = Integer.parseInt(br.readLine());

        Queue<Integer> queue = new ArrayDeque<>();

        for (int i = 1; i < N+1; i++){
            queue.offer(i);
        }

        while (queue.size() > 1){
            queue.poll();
            int last = queue.poll();
            queue.offer(last);
        }

        sb.append(queue.peek());
        System.out.println(sb);
        
    }

}

File: 백준/Silver/2164. 카드2/카드2.py
from collections import deque
import sys

N = int(sys.stdin.readline())

queue = deque()

for i in range(1, N+1):
    queue.append(i)

while True:
    if len(queue) == 1:
        break
    queue.popleft()
    last = queue[0]
    queue.popleft()
    queue.append(last)

print(*queue)